src/rdkafka_buf.c

```diff
@@ -148,9 +154,18 @@ rd_kafka_buf_t *rd_kafka_buf_new_request (rd_kafka_broker_t *rkb, int16_t ApiKey
         /* CorrId: updated later */
         rd_kafka_buf_write_i32(rkbuf, 0);
 
-        /* ClientId: possibly updated later if a flexible version */
+        /* ClientId */
         rd_kafka_buf_write_kstr(rkbuf, rkb->rkb_rk->rk_client_id);
 
+        if (is_flexver) {
+                /* Must set flexver after writing the client id since
+                 * it is still a standard non-compact string. */
+                rkbuf->rkbuf_flags |= RD_KAFKA_OP_F_FLEXVER;
```

Comment:
is this right? from the kip:

> The new RequestHeader version will be version 1, superseding version 0.  In this new version, the RequestHeader's ClientId string will be a COMPACT_STRING rather than  STRING

Reply:
Yeah I don't think that actually materialized, the header ClientId is still a 16-bit lengthed string.